### PR TITLE
Fix the #66 issue. This will ensure that we dont enforce sparsity in approximation

### LINKED ISSUE #66: Better sparsity promotion
Right now sparsity is promoted by an L1 Norm on all the wavelet coefficients in the examoles. However the coarse scale of the wavelet transform has no reason to be sparse and shouldn't be biased.

The examples should highlight that. For this, it might be necessary to have a helper function which for a given `mu` and a given wavelet transform returns the corresponding thresholding levels for all coefficients (the ones corresponding to the coarse scale being 0)

### PR DESCRIPTION
**Pull request by chaithyagr**
Resolves #66 

This is a minor hacky fix, but it should get the job done for now!

While we work on removing flattening, I feel this is a good first step to ensure we don't end enforcing sparsity on coarse-scale!

Another option to set up a regularizer with 0 weights for approximation, but that still keeps computation hard! Not to mention it also has issues that it depends on user to do it.

### CODE REVIEW

File: mri/operators/linear/wavelet.py [hunk 1]
```diff
@@ -125,10 +126,10 @@ def op(self, data):
             )
             coeffs = np.asarray(coeffs)
         else:
-            coeffs, self.coeffs_shape = self._op(data)
+            coeffs, self.coeffs_shape, self.coarse = self._op(data)
         return coeffs
 
-    def _adj_op(self, coeffs, coeffs_shape, dtype="array"):
+    def _adj_op(self, coeffs, coeffs_shape, coarse, dtype="array"):
```
Comment (zaccharieramzi) [inline mri/operators/linear/wavelet.py:132]: don't forget to add it to the docs

File: mri/operators/linear/wavelet.py [hunk 2]
```diff
@@ -125,10 +126,10 @@ def op(self, data):
             )
             coeffs = np.asarray(coeffs)
         else:
-            coeffs, self.coeffs_shape = self._op(data)
+            coeffs, self.coeffs_shape, self.coarse = self._op(data)
```
Comment (zaccharieramzi) [inline mri/operators/linear/wavelet.py:129]: Are there not parts of the code which depend on getting all the coefficients?

**Review by zaccharieramzi — CHANGES_REQUESTED** [review]
One q and one remark

File: mri/operators/linear/wavelet.py (hunk 1 above)
Comment (philouc) [reply mri/operators/linear/wavelet.py:132]: Absolutely zaccharie you're right!

File: mri/operators/linear/wavelet.py (hunk 2 above)
Comment (chaithyagr) [reply mri/operators/linear/wavelet.py:129]: Hmmm not from what I recall. Let me add another method to get the coefficients anyway.
(esp currently we only return flattened versions)

**Comment by chaithyagr** [comment]
Looks like this wont work after all. In optimization we subtract the values of detail, however as we retain the same approximation coefficients, we in the end just mess up the results!
The only way forward is to introduce the weights such that we dont modify the coarse scale!

I am getting that PR up. Closing this here...

**Pull request closed without merging**